src/Reflection/SignatureMap/NativeFunctionReflectionProvider.php

```diff
@@ -137,17 +140,22 @@ public function findFunctionReflection(string $functionName): ?NativeFunctionRef
 						);
 					}
 
-					return new NativeParameterReflection(
+					return new NativeParameterWithPhpDocsReflection(
 						$parameterSignature->getName(),
 						$parameterSignature->isOptional(),
 						TypehintHelper::decideType($type, $phpDocType),
+						$phpDocType ?? new MixedType(),
+						$type,
 						$parameterSignature->passedByReference(),
 						$parameterSignature->isVariadic(),
 						$parameterSignature->getDefaultValue(),
+						$phpDoc !== null ? NativeFunctionReflectionProvider::getParamOutTypeFromPhpDoc($parameterSignature->getName(), $phpDoc) : null,
 					);
 				}, $functionSignature->getParameters()),
 				$functionSignature->isVariadic(),
 				TypehintHelper::decideType($functionSignature->getReturnType(), $phpDoc !== null ? $this->getReturnTypeFromPhpDoc($phpDoc) : null),
```

Comment:
You hacen't changed it but looks like on line 156 we could replace `$this->getReturnTypeFromPhpDoc($phpDoc)` with `$phpDocReturnType`?

Reply:
good catch. this line simplified a lot.